src/H5FDmulti.c

```diff
@@ -2015,6 +2017,52 @@ open_members(H5FD_multi_t *file)
 
     return 0;
 }
+
+/*-------------------------------------------------------------------------
+ * Function:    H5FD_multi_delete
+ *
+ * Purpose:     Delete a file
+ *
+ * Return:      SUCCEED/FAIL
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5FD_multi_delete(const char *filename, hid_t fapl_id)
+{
+    char                     full_filename[H5FD_MULT_MAX_FILE_NAME_LEN];
+    const H5FD_multi_fapl_t *fa;
+    static const char *      func = "H5FD_multi_delete"; /* Function Name for error reporting    */
+
+    /* Clear the error stack */
+    H5Eclear2(H5E_DEFAULT);
+
+    assert(filename);
+
+    /* Quiet compiler */
+    (void)fapl_id;
+
+    /* Get the driver info */
+    fa = (const H5FD_multi_fapl_t *)H5Pget_driver_info(fapl_id);
+    assert(fa);
+
+    /* Delete each member file using the underlying fapl */
+    UNIQUE_MEMBERS (fa->memb_map, mt) {
+        assert(fa->memb_name[mt]);
+        assert(fa->memb_fapl[mt] >= 0);
+
+        /* Note: This truncates the user's filename down to only sizeof(tmp)
+         *      characters. -QK & JK, 2013/01/17
+         */
+        sprintf(full_filename, fa->memb_name[mt], filename);
```

Comment:
Seems like snprintf was intended here?

Reply:
Still concerned about this - it's basically a potential buffer overrun.

Reply:
Should be improved now